middle_end/flambda2/simplify/simplify_expr.ml

```diff
@@ -16,51 +16,45 @@
 
 open! Simplify_import
 
-let simplify_toplevel_common dacc simplify
-    ~(in_or_out_of_closure : Closure_info.in_or_out_of_closure)
-    ~return_continuation ~return_arity ~exn_continuation ~return_cont_scope
-    ~exn_cont_scope =
+let simplify_toplevel_common dacc simplify ~params ~return_continuation
+    ~return_arity ~exn_continuation ~return_cont_scope ~exn_cont_scope =
   (* The usage analysis needs a continuation whose handler holds the toplevel
      code of the function. Since such a continuation does not exist, we create a
      dummy one here. *)
   let dummy_toplevel_cont =
     Continuation.create ~name:"dummy_toplevel_continuation" ()
   in
   let dacc =
-    DA.map_data_flow dacc ~f:(Data_flow.init_toplevel dummy_toplevel_cont [])
+    DA.map_flow_acc dacc ~f:(Flow.Acc.init_toplevel ~dummy_toplevel_cont params)
   in
   let expr, uacc =
     simplify dacc ~down_to_up:(fun dacc ~rebuild ->
         let dacc =
-          DA.map_data_flow dacc
-            ~f:(Data_flow.exit_continuation dummy_toplevel_cont)
+          DA.map_flow_acc dacc
+            ~f:(Flow.Acc.exit_continuation dummy_toplevel_cont)
         in
-        let data_flow = DA.data_flow dacc in
+        let data_flow = DA.flow_acc dacc in
+        let closure_info = DE.closure_info (DA.denv dacc) in
         (* The code_age_relation and used value_slots are only correct at
            toplevel, and they are only necessary to compute the live code ids,
            which are only used when simplifying at the toplevel. So if we are in
            a closure, we use empty/dummy values for the code_age_relation and
            used_value_slots, and in return we do not use the reachable_code_id
            part of the data_flow analysis. *)
-        let code_age_relation, used_value_slots =
-          match in_or_out_of_closure with
-          | In_a_closure -> Code_age_relation.empty, Or_unknown.Unknown
+        let code_age_relation, used_value_slots, print_name =
```

Comment:
Do we keep the changes that were needed due to  `print_name` being used for debug?